src/Arcus.Security.Providers.AzureKeyVault/Configuration/KeyVaultConfiguration.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using Arcus.Security.Providers.AzureKeyVault.Configuration.Interfaces;
+using GuardNet;
+
+namespace Arcus.Security.Providers.AzureKeyVault.Configuration
+{
+    public class KeyVaultConfiguration : IKeyVaultConfiguration
+    {
+        /// <summary>
+        ///     The Uri of the Azure Key Vault you want to connect to.
+        /// </summary>
+        /// <example>https://{vaultname}.vault.azure.net/</example>
+        public Uri VaultUri { get; }
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="vaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
+        /// <exception cref="ArgumentNullException">Exception thrown when the vault is not using https</exception>
+        public KeyVaultConfiguration(Uri vaultUri)
+        {
+            Guard.For<ArgumentException>(() => vaultUri.Scheme != Uri.UriSchemeHttps);
```

Comment:
Or `UriFormatException`? Not blocking of course, just suggestion.

Reply:
`Uri` is a `class` so it can be `null`!

Reply:
Good suggestion :+1: